ideascube/serveradmin/catalog.py

```diff
@@ -126,6 +129,10 @@ def remove(self, package):
         self._removed.append(package)
 
     def commit(self):
+        visible_packages = get_config('ideascube', 'displayed-package-ids')
+        [visible_packages.remove(p.id) for p in self._removed if p.id in visible_packages]
```

Comment:
> The [some(foo) for foo in foos if test(foo)] seems to become the standard way of writing (composition of) map/filter.

That's not becoming the standard, that's exactly how you compose a map and filter. Or rather, it is usually `bars = [some(foo) for foo in foos if test(foo)]` (or reassigning to `foos` if the initial value is not needed any more).

Except that's not what you did.

What you did is `[foos.remove(foo) for foo in bars if foo in foos]`. There are two key differences:

1. See that `bars` in the middle?
2. During the iteration you're modifying the condition being tested

You're iterating over `bars`, applying a function to each item. So you're pretty much doing a `map`. However, the function you apply has a side-effect that modifies another list which turns out to be your `filter`.

And you're doing that in a list-comprehension, not storing the return value, which makes it look at a first glance like it does nothing. (and is honestly just weird, this is the first time I ever see a list comprehension not being assigned to a variable)

> Even Guido suggests to remove map/filter/reduce from python3 (http://www.artima.com/weblogs/viewpost.jsp?thread=98196)

Cool, thanks for the link, I didn't know about his opinion on this. However you'll notice that I didn't suggest using any of these functions, so I don't see how that's relevant. :wink: 

> I think that visible_packages = [p for p in visible_packages if p.id not in self._removed] would be ok for every one (?)

Nope, it would not work, because `visible_packages` contains package ids, not package objects.